### PR TITLE
feat(ui): introduce Project model & co

### PR DESCRIPTION
**Pull request by obdulia-losantos**
# Generic
- `--noinput` option running tests

# CouchDB-Sync
- [x] Fix admin tests

# Kernel
- [x] Project schema serializer: include the linked schema definition

# UI

## backend

- [x] Models: `verbose_name` in  model fields
- [x] Models: remove uniqueness name constraint
- [x] Contract: rename `mapping` to `mapping_rules`
- [x] Contract: add `mapping` field for the kernel mapping id.
- [x] Models: Project model & co
- [x] Serializers: Pipelines without project are automatically bound to default project.
- [x] Contract: read only contracts cannot be updated.
- [x] Fetch from kernel using UI project tree and new fields.
- [x] Publish projects (ensures that th project exists in kernel)
- [x] Publish pipelines (ensures that the mapping set exists in kernel)
- [x] Publish contracts not based in artefacts names.
- [x] Publish contract pre-flight, checks that the linked artefacts... exist in kernel and belong to the same artefacts tree than in UI.
- [x] test, test, test.

## frontend

- [x] cleaning, refactoring, reorganize pieces of code...
- [x] remove `PROJECT_NAME` constant
- [x] take out `kernelUrl` reducer of `constants`
- [x] create `avro-utils` with all AVRO utilities
- [x] make it readable
- [x] clean pipeline redux
- [x] Input: derived schema takes pipeline name
- [x] move identity contract button from input section to contract settings
- [x] fix publish button
- [x] remove `mappingset` in the submissions url, include it only in the `payload`
- [x] test, test, test

### CODE REVIEW

File: aether-ui/aether/ui/api/serializers.py [hunk 1]
```diff
@@ -49,12 +55,47 @@ class PipelineSerializer(DynamicFieldsMixin, serializers.ModelSerializer):
...
+            name=instance.name,  # re-use same name
+            pipeline=instance,
+        )
+
+        return instance
+
     def update(self, instance, validated_data):
         if instance.contracts.filter(is_read_only=True).count() > 0:
-            raise serializers.ValidationError({'description': _('Input is readonly')})
+            raise serializers.ValidationError({'detail': _('Pipeline is read only')})
```
Comment (admbtlr) [inline aether-ui/aether/ui/api/serializers.py:82]: Is this correct? Can a whole Pipeline be read-only? I thought that Inputs and Contracts could be read-only, but that it is always possible to add a new Contract to an existing Pipeline, even if the Input is read-only?

Comment (obdulia-losantos) [reply aether-ui/aether/ui/api/serializers.py:82]: You can always add a new contract, what you cannot do is to edit the pipeline, I mean the input JSON data and the AVRO schema.

File: aether-ui/aether/ui/api/tests/test_utils.py [hunk 2]
```diff
@@ -16,16 +16,467 @@
...
+        outcome = utils.publish_preflight(contract)
+        self.assertEqual(
+            outcome,
+            {
+                'error': [
+                    'Pipeline has no input',
+                    'Pipeline belongs to a different project in kernel',
+                    'Contract belongs to a different project in kernel',
+                ],
+                'warning': [
```
Comment (admbtlr) [inline aether-ui/aether/ui/api/tests/test_utils.py:400]: Seeing them listed out here, I feel like `warning` and `info` should be switched. It looks like the `warning` entries are informing the user about what is going to happen, while the `info` entries are warning the user about the implications of it happening.

File: aether-ui/aether/ui/assets/apps/pipeline/sections/Input.jsx [hunk 3]
```diff
@@ -21,105 +21,45 @@
...
-    id: 'pipeline.input.schema.invalid.message.head.recursive'
+    id: 'input.schema.invalid.message.head.recursive'
   },
   regularError: {
     defaultMessage: 'You have provided an invalid AVRO schema.',
-    id: 'pipeline.input.schema.invalid.message.head'
+    id: 'input.schema.invalid.message.head'
+  },
+  inputDataPlaceholder: {
+    defaultMessage: 'We will generate some sample data for you, once you have added a schema. Or, Add data in JSON Format and Aether will derive an AVRO schema for you.',
```
Comment (admbtlr) [inline aether-ui/aether/ui/assets/apps/pipeline/sections/Input.jsx]:
```suggestion
    defaultMessage: 'Once you have added a schema, we will generate some sample data for you. Alternatively, you can enter some JSON data and Aether will derive an AVRO schema for you.',
```

**Review by admbtlr — COMMENTED** [review]
Amazing work! Love the clarity! Finished the code review, just need to do a quick QA check.

**Review by shawnsarwar — COMMENTED** [review]
I can't believe you did all this in less than a week! :1st_place_medal:

File: aether-ui/aether/ui/api/tests/test_utils.py (hunk 2 above)
Comment (obdulia-losantos) [reply aether-ui/aether/ui/api/tests/test_utils.py:400]: What do you prefer to do in this case?

Comment (admbtlr) [reply aether-ui/aether/ui/api/tests/test_utils.py:400]: So in this case I would just switch them around - all warnings should become info and vice versa.

Comment (obdulia-losantos) [reply aether-ui/aether/ui/api/tests/test_utils.py:400]: To confirm that I understood it correctly...

## Now

### Error (blockers, with any the contract cannot be published)

- Pipeline has no input
- Pipeline belongs to a different project in kernel
- Contract is read only
- Contract has no entity types
- Contract has no mapping rules
- Contract mapping rules have errors
- Contract belongs to a different project in kernel
- Contract belongs to a different pipeline (mapping set) in kernel
- Entity type "XXX" (as project schema) belongs to a different project in kernel

### Warning (warns about the publish effects)

- Project will be published
- Pipeline (as mapping set) will be published
- Input data will be changed
- Schema data will be changed
- Contract (as mapping) will be published
- Mapping rules data will be changed
- Entity type "XXX" (as schema) will be published
- Entity type "XXX" (as schema) data will be changed
- Entity type "XXX" (as project schema) will be published

### Info (indicates that part of the artefacts were already published)

- Project is already published
- Pipeline (as mapping set) is already published
- Contract (as mapping) is already published
- Entity type "XXX" (as schema) is already published
- Entity type "XXX" (as project schema) is already published

## After

### Error (blockers, with any the contract cannot be published)

- Pipeline has no input
- Pipeline belongs to a different project in kernel
- Contract is read only
- Contract has no entity types
- Contract has no mapping rules
- Contract mapping rules have errors
- Contract belongs to a different project in kernel
- Contract belongs to a different pipeline (mapping set) in kernel
- Entity type "XXX" (as project schema) belongs to a different project in kernel

### Warning (warns if any of the artefacts was already published)

- Project is already published
- Pipeline (as mapping set) is already published
- Contract (as mapping) is already published
- Entity type "XXX" (as schema) is already published
- Entity type "XXX" (as project schema) is already published

### Info (informs about the publish consequences)

- Project will be published
- Pipeline (as mapping set) will be published
- Input data will be changed
- Schema data will be changed
- Contract (as mapping) will be published
- Mapping rules data will be changed
- Entity type "XXX" (as schema) will be published
- Entity type "XXX" (as schema) data will be changed
- Entity type "XXX" (as project schema) will be published

@adamvert is this what you want?

**Review by admbtlr — APPROVED** [review]
🥇

**Pull request merged**